fissa/tests/generate_tiffs.py

```diff
@@ -0,0 +1,120 @@
+        (expected / 10, "float32", None),
+        (expected / 10, "float64", None),
+    ):
+        if dtype.startswith("uint"):
+            data = np.abs(data)
+        if dtype:
+            data = data.astype(dtype)
+        if shp:
+            data = data.reshape(shp)
+
+        qualifier = ""
+        if shp:
+            qualifier += "_" + ",".join(str(x) for x in shp)
+        qualifier += "_{}".format(dtype)
+
+        if not shp:
+            imageio.imwrite(
+                os.path.join(output_dir, "imageio.imwrite{}.tif".format(qualifier)),
+                data[0],
+            )
+
+        tifffile.imsave(
+            os.path.join(output_dir, "tifffile.imsave{}.tif".format(qualifier)),
+            data,
+        )
+        tifffile.imsave(
+            os.path.join(output_dir, "tifffile.imsave.bigtiff{}.tif".format(qualifier)),
+            data,
+            bigtiff=True,
+        )
+        if dtype in ("uint8", "uint16", "float32"):
+            tifffile.imsave(
+                os.path.join(
+                    output_dir, "tifffile.imsave.imagej{}.tif".format(qualifier)
+                ),
+                data,
+                imagej=True,
+            )
+
+        with tifffile.TiffWriter(
+            os.path.join(
+                output_dir, "TiffWriter.write.discontiguous{}.tif".format(qualifier)
+            )
+        ) as tif:
+            for frame in data:
+                tif.write(frame, contiguous=False)
+
+        with tifffile.TiffWriter(
+            os.path.join(
+                output_dir, "TiffWriter.write.contiguous{}.tif".format(qualifier)
+            )
+        ) as tif:
+            for frame in data:
+                tif.write(frame, contiguous=True)
+
+        with tifffile.TiffWriter(
+            os.path.join(output_dir, "TiffWriter.save{}.tif".format(qualifier))
+        ) as tif:
+            for frame in data:
+                tif.save(frame)
+
+        if not shp:
+            with tifffile.TiffWriter(
+                os.path.join(output_dir, "TiffWriter.mixedA{}.tif".format(qualifier))
+            ) as tif:
+                tif.write(data[:2], contiguous=False)
+                tif.write(data[2:4], contiguous=True)
+                tif.write(data[4:], contiguous=False)
+
+        if not shp:
+            with tifffile.TiffWriter(
+                os.path.join(output_dir, "TiffWriter.mixedB{}.tif".format(qualifier))
+            ) as tif:
+                tif.write(data[:4], contiguous=False)
+                tif.write(data[4:], contiguous=False)
+
+        if not shp:
+            with tifffile.TiffWriter(
+                os.path.join(output_dir, "TiffWriter.mixedC{}.tif".format(qualifier))
+            ) as tif:
+                tif.write(data[:2], contiguous=True)
+                tif.write(data[2:4], contiguous=False)
+                tif.save(data[4:])
+
+
+if __name__ == "__main__":
+    print("Using imageio {}".format(imageio.__version__))
```

Comment:
We should print the used package versions into a text-file so it's clear which version was used for the actual tiffs stored on the repo now.

Reply:
If you look at the commit (120784d) that introduced/last modified these test resources (which is an obvious thing to do if you are trying to check the provenance of the files), it says the version numbers in the commit message. I don't think we need it in a text file. To me at least, it's not obvious to expect that there should be a text file with this content to look for, and it will easily be lost in amongst a directory of 92 tiff files.

Reply:
Fair enough, happy to not have that.